### PR TITLE
Fix Decrypt message via occ

### LINKED ISSUE #23695: When trying to decrypt a single user using occ it states to decrypt ALL files
### Steps to reproduce
1. Admin wants to decrypt a single user by issuing the command `sudo -u www-data php occ encryption:decrypt user`

### Expected behaviour
Only user should be decrypted without disabling the encryption backend for all the other users. The message should be unambigously telling you that only the selected user gets decrypted and act accordingly (i. e. not decrypting everything and disabling the encryption backend).

### Actual behaviour
The cli tool occ tells you this:

```
Disable server side encryption... done.

You are about to start to decrypt all files stored in your ownCloud.
It will depend on the encryption module and your setup if this is possible.
Depending on the number and size of your files this can take some time
Please make sure that no user access his files during this process!

Do you really want to continue? (y/n) 
```

### Server configuration
**Operating system**: Debian 8 "Jessie"

**Web server:** Apache 2.4.10 (from default repository)

**Database:** Mysql 5.5.46 (from default repository)

**PHP version:** 5.6.14 (from default repository)

**ownCloud version:** 8.2.3

**Updated from an older ownCloud or fresh install:** 8.1

**List of activated apps:**
Enabled:
  - activity: 2.1.3
  - bookmarks: 0.6
  - calendar: 0.8.1
  - contacts: 0.5.0.0
  - documents: 0.11.0
  - encryption: 1.1.0
  - external: 1.2
  - files: 1.2.0
  - files_clipboard: 0.4.1
  - files_external: 0.3.0
  - files_pdfviewer: 0.7
  - files_sharing: 0.7.0
  - files_texteditor: 2.0
  - files_versions: 1.1.0
  - files_videoviewer: 0.1.3
  - firstrunwizard: 1.1
  - gallery: 14.2.0
  - imprint: 0.2.5
  - notifications: 0.1.0
  - provisioning_api: 0.3.0
  - search_lucene: 0.7.0
  - templateeditor: 0.1
  - files_trashbin
Disabled:
  - updater
  - user_external
  - user_ldap

**The content of config/config.php:**
{
    "system": {
        "instanceid": "oc8bj094mu3d",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***""
        ],
        "datadirectory": "\/var\/www\/owncloud\/data",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***"",
        "dbtype": "mysql",
        "version": "8.2.1.4",
        "dbname": "owncloud",
        "dbhost": "127.0.0.1",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "forcessl": true,
        "forceSSLforSubdomains": true,
        "theme": "",
        "maintenance": false,
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "loglevel": 0,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "ldapIgnoreNamingRules": false,
        "appstore.experimental.enabled": true,
        "trashbin_retention_obligation": "auto",
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "timeout": 0
        }
    }
}

**Are you using encryption:** yes

### PR DESCRIPTION
**Pull request by davitol**
Fixes https://github.com/owncloud/core/issues/23695

### CODE REVIEW

File: core/Command/Encryption/DecryptAll.php [hunk 1]
```diff
@@ -140,6 +149,7 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 				$result = $this->decryptAll->decryptAll($input, $output, $user);
 				if ($result === false) {
 					$output->writeln(' aborted.');
+					$output->write('Enable server side encryption... ');
```
Comment (nickvergessen) [inline core/Command/Encryption/DecryptAll.php]: writeln please

File: core/Command/Encryption/DecryptAll.php [hunk 2]
```diff
@@ -127,8 +129,15 @@ protected function execute(InputInterface $input, OutputInterface $output) {
...
+			$uid = $input->getArgument('user');
+			if (empty($uid)) {
+				$message = 'your ownCloud';
+			} else {
+				$message = "$uid's account";
+			}
+
 			$output->writeln("\n");
-			$output->writeln('You are about to start to decrypt all files stored in your ownCloud.');
+			$output->writeln("You are about to start to decrypt all files stored in $message.");
```
Comment (nickvergessen) [inline core/Command/Encryption/DecryptAll.php]: string concatenation is bad for translating

Comment (nickvergessen) [inline core/Command/Encryption/DecryptAll.php]: ah we dont translate occ stuff, right

File: core/Command/Encryption/DecryptAll.php [hunk 3]
```diff
@@ -118,6 +118,8 @@ protected function configure() {
 	protected function execute(InputInterface $input, OutputInterface $output) {
 
 		try {
+			
+
```
Comment (SergioBertolinSG) [inline core/Command/Encryption/DecryptAll.php]: Not needed white lines.

**Comment by davitol** [comment]
https://github.com/owncloud/core/blob/eaeb5d606dee28f647f2f2b34ec49c9cb6c091ec/core/Command/Encryption/DecryptAll.php#L133 this line should be fix too when https://github.com/owncloud/core/issues/24994 check will be solved

**Comment by PVince81** [comment]
@davitol did you try changing this check instead https://github.com/owncloud/core/blob/eaeb5d606dee28f647f2f2b34ec49c9cb6c091ec/lib/private/Encryption/DecryptAll.php#L144 ?

Or is a zero user causing issues in more pieces of the code ?

File: core/Command/Encryption/DecryptAll.php [hunk 4]
```diff
@@ -127,8 +127,16 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 				return;
 			}
 
+			$uid = $input->getArgument('user');
+			//FIXME WHEN https://github.com/owncloud/core/issues/24994 is fixed
+			if (empty($uid)) {
```
Comment (nickvergessen) [inline core/Command/Encryption/DecryptAll.php]:
```php
if ($uid === null) {
```
to allow user name `'0'` to act correctly.

File: core/Command/Encryption/DecryptAll.php [hunk 5]
```diff
@@ -140,6 +148,7 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 				$result = $this->decryptAll->decryptAll($input, $output, $user);
 				if ($result === false) {
 					$output->writeln(' aborted.');
+					$output->writeln('Enable server side encryption... ');
```
Comment (rullzer) [inline core/Command/Encryption/DecryptAll.php]: Probabaly should be something like:

`Server side encryption remains enabled`

**Comment by nickvergessen** [comment]
Needs rebase after https://github.com/owncloud/core/pull/24998 is merged

**Comment by SergioBertolinSG** [comment]
I'll take this over.

**Comment by nickvergessen** [comment]
:+1:

**Comment by guruz** [comment]
👍

**Comment by DeepDiver1975** [comment]
@davitol please prepare a pull request for stable9 holding these changes - no idea if #25022 and #24998 are required as well - THX

**Comment by DeepDiver1975** [comment]
I'll take care about the backport

**Comment by PVince81** [comment]
stable9: https://github.com/owncloud/core/issues/25188

**Pull request merged**